### PR TITLE
Implement Wayland-compatible eyedropper using XDG Desktop Portals

### PR DESCRIPTION
**Pull request by filip-hejsek**
This PR adds a new XDG Desktop Portal-based eyedropper implementation for Linux.

The Portal-based dropper has the following limitations:
 - It doesn't work if Desktop Portals are not properly installed or configured on the system (this may affect users on old or less maintained distros, those using more obscure desktop environments, or users with custom setups).
 - The functionality is not equivalent to the current dropper: The current dropper captures a 7x7 area around cursor, but the Portal API only gives us the color directly under cursor.

For the vast majority of users on Linux, this is an improvement over the current dropper, which:
 - is broken on wxGTK3 with X11 (the color picker works exactly once, i.e. if you pick the color a second time, the screenshot it takes is just the same as the first one until you restart Aegisub).
 - doesn't work on Wayland at all, not even XWayland (this is unfixable because of Wayland security model).

Implementation notes:
 - I added some new files so I had to put some copyright notice there. I copied the notice from other files, but changed the copyright statement to `Copyright (c) 2026, Aegisub contributors` and updated the project URL to `https://aegisub.org/`. I noticed the copyright notices in most files are incorrect (they were not updated when the file was modified).
 - I put some things I added into a new `agi::xdp_utils` namespace. Not sure if this matches current conventions, although existing use of namespaces seemed inconsistent to me anyway.
 - I added an `#ifdef` around `xdg_desktop_portal_utils.cpp`, even though it's unnecessary because the file is compiled conditionally. This matches how it's done in the `audio_player_*` files.
 - I remember encountering off-by-one errors (where using the eyedropper on the current color preview would give a slightly different color), but I was unable to reproduce it.
 - If invoking the color picker fails, no feedback is provided (nothing happens when the button is clicked). Displaying an error message would be better.
 - There is a (very unlikely) possibility that the dialog could be closed before the OS returns the color from the Portal API. This case is not handled correctly in the code, it just sends an event to a dangling pointer.

### CODE REVIEW

File: src/dialog_colorpicker.cpp [hunk 1]
```diff
@@ -1100,6 +1136,49 @@ void DialogColorPicker::OnCaptureLost(wxMouseCaptureLostEvent&) {
...
+	return static_cast<unsigned char>(x * 255 + 0.5);
+	// NOTE: I have encountered cases where the returned value was off by one,
+	//       probably due to incorrect rounding. However, I am unable to reproduce it.
+}
+
+static void PortalPickColorCallback(GObject *source, GAsyncResult *res, gpointer data) {
+	// this will be called from GLib main loop, which likely runs on wxWidgets main thread, but don't rely on it
+	wxEvtHandler *event_target = static_cast<wxEvtHandler *>(data);
+	GError *error = nullptr;
+	GVariant *color_result = xdp_portal_pick_color_finish(XDP_PORTAL(source), res, &error);
```
Comment (arch1t3cht) [inline src/dialog_colorpicker.cpp]: Similarly to the `*portal` comment, `error` and `color_result` should be written into `std::unique_ptr`s with custom deleters to automatically handle the freeing:

```c++
GError *raw_error = nullptr;
std::unique_ptr<GVariant, decltype(&g_variant_unref)> color_result{xdp_portal_pick_color_finish(XDP_PORTAL(source), res, &raw_error), g_variant_unref};    // FIXME: Use out_ptr here once we can use C++23
std::unique_ptr<GError, decltype(&g_error_free)>{raw_error, g_error_free};
// [...]
```

**Review by arch1t3cht — CHANGES_REQUESTED** [review]
Thanks a lot for working on this! This has been a major annoyance for a long time, but I always thought that it'd require a lot more custom logic to fix.

Replying to some of your implementation notes:

> I added some new files so I had to put some copyright notice there. I copied the notice from other files, but changed the copyright statement to `Copyright (c) 2026, Aegisub contributors`

1. Of course you're welcome to add your own name here if you like, but of course you can also keep it as `Aegisub contributors`.
2. Just like most other parts of Aegisub, its license situation is a bit of a mess, but from my investigations it looks like the BSD 3-clause license was used for all source files in the first few years of the project, but for a while now (since around 2010-2011, it seems, so for around 3 years before the project started fizzling out for a while) the [OpenBSD license](https://en.wikipedia.org/wiki/ISC_license#OpenBSD_license) (i.e. the ISC license without the `and/or` wording; you can find an example in files like `libaegisub/common/vfr.cpp`) has been used for newer files and the previous maintainer tgoyne has replaced BSD3 license headers with OpenBSD license headers whenever he significantly rewrote a source file.

    So, you're of course welcome to keep the license as BSD3 if you strongly prefer it, but if you don't have any strong opinions on this it would be more consistent to continue the previous trend and use the OpenBSD license for new files.
    (Afaik this would be the first new source files added to upstream since TypesettingTools officially took over maintainership, so there's no official policy on this yet.)

> I put some things I added into a new agi::xdp_utils namespace. Not sure if this matches current conventions, although existing use of namespaces seemed inconsistent to me anyway.

Sure, that's good. The existing use of namespaces is indeed inconsistent, but for new code using namespaces is definitely better than not using them.

> There is a (very unlikely) possibility that the dialog could be closed before the OS returns the color from the Portal API. This case is not handled correctly in the code, it just sends an event to a dangling pointer.

Thanks for pointing this out. Do you know if `xdp_portal_pick_color` guarantees that the passed callback will always be called? The docs seem to suggest this, but they're not completely explicit about it. Either way, I think this can be handled using some `shared_ptr`/`weak_ptr` shenanigans. I can try to work something out regarding this (though without knowing a scenario where this edge case happens it'll also be hard to test the handling for it.)

---

For the rest, see the individual review comments. Mostly some small nits and asking to use `std::unique_ptr`s to manage ownership. If you'd like me to implement those changes myself, let me know.

File: meson.build [hunk 2]
```diff
@@ -232,6 +231,13 @@ foreach dep: [
     ['hunspell', [], 'Hunspell', ['hunspell', 'hunspell_dep'], []],
     ['uchardet', [], 'uchardet', ['uchardet', 'uchardet_dep'], []],
 ]
+
+if host_machine.system() == 'linux'
```
[the remark this answers is not in the document]
Comment (arch1t3cht) [reply meson.build]: I see, thanks for explaining

File: src/dialog_colorpicker.cpp (hunk 1 above)
Comment (filip-hejsek) [reply src/dialog_colorpicker.cpp]: done

File: meson.build (hunk 2 above)
[the remark this answers is not in the document]
Comment (filip-hejsek) [reply meson.build]: I have decided to change it to add `libportal-gtk3` if `__WXGTK3__` is defined, and just `libportal` otherwise.

**Comment by filip-hejsek** [comment]
So the build failed on Ubuntu:

```
In file included from ../src/xdg_desktop_portal_utils.cpp:27:
/usr/include/libportal-gtk3/portal-gtk3.h:23:10: fatal error: gtk/gtk.h: No such file or directory
   23 | #include <gtk/gtk.h>
      |          ^~~~~~~~~~~
```

I guess we need `gtk3-dev` or something, but shouldn't that be a dependency of `libportal-gtk3-dev`?

**Comment by filip-hejsek** [comment]
Somehow the compilation flags for `xdg_desktop_portal_utils.cpp` don't contain the necessary include directories, even though `pkgconf --cflags libportal-gtk3` includes them.

**Review by arch1t3cht — APPROVED** [review]
